src/librustc_mir/interpret/memory.rs

```diff
@@ -462,10 +462,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
                         // no need to report anything, the const_eval call takes care of that
                         // for statics
                         assert!(tcx.is_static(def_id));
-                        match err {
-                            ErrorHandled::Reported => err_inval!(ReferencedConstant),
-                            ErrorHandled::TooGeneric => err_inval!(TooGeneric),
-                        }
+                        err
```

Comment:
I am a bit confused, why does this typecheck? Looks like `err` has type `ErrorHandled`, not `InterpError`.

Reply:
`?` calls `Into::into` (or `From::from`, one of them) on the error ;)

Reply:
Ah, I had no idea we had a `impl From<ErrorHandled> for InterpErrorInfo<'_>`. :D